app.py

```diff
@@ -0,0 +1,67 @@
+import cv2
+import streamlit as st
+from deepface import DeepFace
+import tempfile
+
+face_cascade = cv2.CascadeClassifier(cv2.data.haarcascades + 'haarcascade_frontalface_default.xml')
+
+def analyze_emotions(frame):
+    gray_frame = cv2.cvtColor(frame, cv2.COLOR_BGR2GRAY)
+    rgb_frame = cv2.cvtColor(gray_frame, cv2.COLOR_GRAY2RGB)
+
+    faces = face_cascade.detectMultiScale(gray_frame, scaleFactor=1.1, minNeighbors=5, minSize=(30, 30))
+
+    for (x, y, w, h) in faces:
+        face_roi = rgb_frame[y:y + h, x:x + w]
+
+        result = DeepFace.analyze(face_roi, actions=['emotion'], enforce_detection=False)
+
+        emotion = result[0]['dominant_emotion']
+
+        cv2.rectangle(frame, (x, y), (x + w, y + h), (0, 0, 255), 2)
+        cv2.putText(frame, emotion, (x, y - 10), cv2.FONT_HERSHEY_SIMPLEX, 0.9, (0, 0, 255), 2)
+
+    return frame
+
+def main():
+    st.title("Real-time Emotion Detection")
```

Comment:
If you plan to make a Streamlit UI on top of my implementation, please implement separation of concern.
Try to keep the code decentralized